skarab/tut_intro/skarab_tut_intro.py

```diff
@@ -0,0 +1,127 @@
+#!/bin/env ipython
+
+'''
+This script demonstrates:
+- programming an FPGA
+- feeding the adder with inputs and checking back the results
+- controlling and monitoring the counter.
+
+Designed for use with tut_intro at the 2019 CASPER workshop.
+\n\n 
+Author: Cedric Viou, 2019.
+'''
+import casperfpga, time, struct, sys, logging, socket, numpy
+
+
+fpgfile = 'skarab_tut_intro.fpg'
+
+
+def exit_fail():
+    logger.fatal('FAILURE DETECTED.' )
+    sys.exit()
+
+def exit_clean():
+    sys.exit()
+
+if __name__ == '__main__':
+    from optparse import OptionParser
+
+    p = OptionParser()
+    p.set_usage('skarab_tut_intro.py <CASPER_FPGA_HOSTNAME_or_IP> [options]')
+    p.set_description(__doc__)
+    p.add_option('', '--noprogram', dest='noprogram', action='store_true',
+        help='Don\'t programm FPGA')  
+    p.add_option('-b', '--fpgfile', dest='fpg', type='str', default=fpgfile,
+        help='Specify the fpg file to load')  
+    opts, args = p.parse_args(sys.argv[1:])
+
+    if args==[]:
+        print('Please specify a CASPER FPGA board. \nExiting.')
+        sys.exit()
+    else:
+        casper_fpga = args[0]
+    if opts.fpg != '':
+        fpgfile = opts.fpg
+try:
+
+    logger = logging.getLogger(casper_fpga)
```

Comment:
As is, this logger doesn't have any handlers to log against, so no log messages are displayed to the console.
This logger is also unnecessary as casperfpga creates a logger against the CasperFpga object with the relevant/necessary handlers.

I would remove this line and only pass the `log_level='INFO'` kwarg.

Reply:
There is an option, when interfacing directly with the `logging` module, to set it up for 'basic configuration' - i.e. `logging.basicConfig()`
- This is usually the reason for duplicate log messages, and this might be present in another config file (or environment variable?)

Either way, fret not. I will make these changes and push to the workshop2019 branch here. Thanks again!

Reply:
Thanks for the explanation.
Modify logger according to what **you** think is best.